driver/src/driver/stablex_driver.rs

```diff
@@ -155,7 +155,8 @@ impl<'a> StableXDriver for StableXDriverImpl<'a> {
             Some(time_limit) => time_limit,
             None => {
                 self.metrics.auction_skipped(batch_to_solve);
```

Comment:
I might have missed this in an earlier PR but this will cause us to log a successful processing (which is not true as we didn't attempt to solve this batch). Maybe we should rename the method to something like `processed_but_not_submitted` to make this clearer.

Reply:
I noticed this as well but wanted to tackle this in a separate PR as I think it requires a bit of though. Right now we have some inconsistencies with how we handle timeouts:
- If the scheduler runs at the end of a batch past the `solver_time_limit`, it will silently (as in no metrics recorded) wait until the next batch
- If the scheduler runs at the end of a batch near the `solver_time_limit` then a couple things can happen:
  1. The orderbook retrieval is in time, but the solver is started with a very low time limit, the solver will probably not find a solution and it will increment the `auction_skipped` metric
  2. The orderbook retrieval puts us over the time limit, so we skip the solving and it will increment a `auction_skipped` metric.

To m,e both cases when the driver attempts to solve at the end of the batch, whether it starts the solver with a very low time limit (for arguments sake, lets say 0.1s) or skips it is about the same. 

A problem with this approach is that we don't really distinguish the case where because of some unforeseen issue, the orderbook retrieval takes, for argument's sake, 3 minutes and causes is to not run the solver. This should be an error but is currently equivalent to the second case mentioned above.

Also, the way we submit metrics is getting a little hard to follow for me, with many branches. I wanted to take a little time to clean this up and maybe address your concerns in the process.

Reply:
Let me know if you are OK with me addressing this in a separate PR.

Reply:
Yes, let's do that in a separate PR. Maybe we should re-introduce a minimum time in which we don't start the solver? We are also running into somewhat benign failures on rinkeby today. 

When we start fetching the orderbook with 3s remaining we expectedly run over time and still log an error.